### PR TITLE
DI injection via method signature

### PR DESCRIPTION
**Pull request by mdmunir**
#9785

### CODE REVIEW

**Comment by SilverFire** [comment]
Code style must be fixed

File: tests/framework/di/ContainerTest.php [hunk 1]
```diff
@@ -100,4 +104,103 @@ public function testDefault()
...
+
+    public function testInvoke()
+    {
+        $this->mockApplication([
+            'components'=>[
+                'qux' => [
+                    'class' => 'yiiunit\framework\di\stubs\Qux',
+                    'a' => 'belongApp',
+                ]
+            ]
```
Comment (SilverFire) [inline tests/framework/di/ContainerTest.php]: We need a test for the case, when we have two components of the same class with different names:

```php
                'qux1' => [
                     'class' => 'yiiunit\framework\di\stubs\Qux',
                     'a' => 'belongAppQux1',
                 ]
```

and the following tests of arguments:

```php
function (stubs\QuxInterface $qux1)
```

```php
function (stubs\QuxInterface $fooName)
```

**Comment by SilverFire** [comment]
@mdmunir don't implement my comments for now, please.
We are discussing the design possibility of merging this PR with the core team.

**Comment by mdmunir** [comment]
What tools that you used to fix code style?

**Comment by SilverFire** [comment]
PHPStorm :)

https://github.com/yiisoft/yii2/blob/master/docs/internals/core-code-style.md

Don't spend your time on this PR until the idea is under discussion in core, please.

**Comment by samdark** [comment]
@mdmunir We'd like to keep `invoke()` but remove all the calls to it. The reason is to keep code easier to debug and read.

**Comment by mdmunir** [comment]
:+1:

**Comment by mdmunir** [comment]
i create new one.

**Pull request closed without merging**